notebooks/utils.py

```diff
@@ -0,0 +1,97 @@
+import json
+
+import pandas as pd
+import requests
+from schemas.jobs import JobResultDownloadResponse
+
+MODEL_INFO_FILE = "assets/model_info.csv"
+
+EVAL_METRICS = {
+    "Meteor": ["meteor", "meteor_mean"],
+    "BERT Precision": ["bertscore", "precision_mean"],
+    "BERT Recall": ["bertscore", "recall_mean"],
+    "BERT F1": ["bertscore", "f1_mean"],
+    "ROUGE-1": ["rouge", "rouge1_mean"],
+    "ROUGE-2": ["rouge", "rouge2_mean"],
+    "ROUGE-L": ["rouge", "rougeL_mean"],
+}
+
+
+def _parse_model_results(result: dict, mi: pd.DataFrame):
+    row = {}
+
+    model_name = result["model"]
+    row["Model"] = model_name
+
+    for column, metric in EVAL_METRICS.items():
+        temp_result = result
+        for key in metric:
+            value = temp_result.get(key)
+            if value is None:
+                break
+            temp_result = value
+
+        row[column] = value
+
+    row["RAM_GB"] = mi[mi.model_name == model_name]["RAM_GB"].to_numpy()[0]
+
+    return row
+
+
+def _download_text_file(result: JobResultDownloadResponse) -> str:
+    download_url = result.download_url
+    download_response = requests.get(download_url)
```

Comment:
I think we can do this directly via the SDK now rather than requests https://github.com/mozilla-ai/lumigator/blob/b2d760b13b6b8569c0a80e1252528e0b62edc7a6/lumigator/python/mzai/sdk/sdk/jobs.py#L175

Reply:
The SDK returns the download URL. This is what I'm using in the Notebook. This function retrieves the actual results.